fabtests/benchmarks/benchmark_shared.c

```diff
@@ -306,16 +307,23 @@ int bandwidth(void)
 				if (ret)
 					return ret;
 			}
+
+			if (opts.fi_more) {
+				if (j < opts.window_size - 1 &&
+				    i >= opts.warmup_iterations &&
+				    i < opts.iterations + opts.warmup_iterations - 1) {
+					flags |= FI_MORE;
+				} else {
+					flags &= ~FI_MORE;
+				}
+			}
```

Comment:
This only applies to the else clause below.  This is starting to nest fairly deep, so a function for this check may be worthwhile.  Also, the check is repeated in the next patch, and the check itself is a little complicated in order to be correct.